core/src/main/java/org/apache/carbondata/core/metadata/datatype/ArrayType.java

```diff
@@ -31,6 +31,31 @@ public boolean isComplexType() {
     return true;
   }
 
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null) {
+      return false;
+    }
+    if (!(obj instanceof ArrayType)) {
+      return false;
+    }
+    if (!this.getName().equalsIgnoreCase(((ArrayType) obj).getName())) {
```

Comment:
Equals and hash should be implemented for all datatypes.Each required to have different logic
1) Nestedtype, should also include childType.equals.
2) Decimal type should consider precision, scala  etc.